### PR TITLE
[WIP] New audio player

### PR DESCRIPTION
**Pull request by mahmoudfelfel**
I'm working on updating the current app's audio player with a new one that has background play capability and supports both iOS and Android.
The main purpose is the audio player, but I think have to do some refactoring on the way, though will try to keep it limited in this PR to get a release out soon.

These are the main tasks included in this PR, there are few small fixes here and there as well:
- [x]  Upgrade react-native, react, react navigation, and native-base.
- [x]  Replace current audio player with react-native-track-player.
- [x]  integrate redux with player state and dispatch events from the track player event handler, so it can work with remote/background events.
- [x]  Change the player buttons to handle the track player events and update the UI accordingly.
- [x]  add a new audio progress bar.

### CODE REVIEW

**Comment by thabti** [comment]
this looks promising - I will look into this tonight iA.

**Comment by thabti** [comment]
let me know when you want a full review.

**Comment by mahmoudfelfel** [comment]
@sabeurthabti will do, thanks :)

**Comment by mahmoudfelfel** [comment]
@sabeurthabti please check.
I think the only thing missing to release the app is to have a splash screen and an appicon, will open another PR for that.

**Review by thabti — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -16,37 +16,39 @@
...
+    "react-navigation": "^2.9.3",
+    "react-navigation-redux-helpers": "^2.0.5",
+    "react-redux": "5.0.7",
+    "redux": "4.0.0",
+    "redux-thunk": "2.3.0",
+    "styled-components": "^3.3.0",
     "zero-fill": "^2.2.3"
   },
   "devDependencies": {
+    "@babel/core": "^7.0.0-beta.55",
```
Comment (thabti) [inline package.json:38]: official release is now out

**Comment by thabti** [comment]
@mahmoudfelfel looks good lets get this merged in.

**Pull request merged**